src/plugins/BasebandRIL.js

```diff
@@ -0,0 +1,312 @@
+            this.networkOperatorName.value = values.slice(2).join(' ');
+        }
+        if (values[0] === 'sim' && values[1] === 'operator') {
+            this.simOperatorMMC.value = values[2];
+        }
+        if (values[0] === 'sim' && values[1] === 'operator_name') {
+            this.simOperatorName.value = values.slice(2).join(' ');
+        }
+        if (values[0] === 'sim' && values[1] === 'imsi_id') {
+            this.simMSIN.value = values[2];
+        }
+        if (values[0] === 'sim' && values[1] === 'phone_number') {
+            this.simOperatorPhoneNumber.value = values[2];
+        }
+    }
+
+    /**
+     * Add the button to the player toolbar.
+     */
+    renderToolbarButton() {
+        const toolbars = this.instance.getChildByClass(this.instance.root, 'gm-toolbar');
+        if (!toolbars) {
+            return; // if we don't have toolbar, we can't spawn the widget
+        }
+
+        const toolbar = toolbars.children[0];
+        this.toolbarBtn = document.createElement('li');
+        this.toolbarBtnImage = document.createElement('div');
+        this.toolbarBtnImage.className = 'gm-icon-button gm-sim-button';
+        this.toolbarBtnImage.title = this.i18n.NETWORK_TITLE || 'Baseband';
+        this.toolbarBtn.appendChild(this.toolbarBtnImage);
+        this.toolbarBtn.onclick = this.toggleWidget.bind(this);
+        toolbar.appendChild(this.toolbarBtn);
+    }
+
+    /**
+     * Render the widget.
+     */
+    renderWidget() {
+        // Create elements
+        this.widget = document.createElement('div');
+        this.form = document.createElement('form');
+
+        // Generate title
+        const title = document.createElement('div');
+        title.className = 'gm-title';
+        title.innerHTML = this.i18n.NETWORK_TITLE || 'Baseband';
+        this.form.appendChild(title);
+
+        if (this.basebandEnabled) {
+            this.networkOperator = document.createElement('div');
+
+            const networkOperatorLabelDiv = document.createElement('div');
+            networkOperatorLabelDiv.className = 'gm-section';
+            const networkOperatorLabel = document.createElement('label');
+            networkOperatorLabel.innerHTML = this.i18n.NETWORK_OPERATOR || 'Network Operator';
+            networkOperatorLabelDiv.appendChild(networkOperatorLabel);
+
+            const networkOperatorMMCDiv = document.createElement('div');
+            networkOperatorMMCDiv.className = 'gm-fields';
+            const networkOperatorMMCLabel = document.createElement('label');
+            networkOperatorMMCLabel.innerHTML = 'MCC/MNC';
+            this.networkOperatorMMC = document.createElement('input');
+            this.networkOperatorMMC.className = 'gm-network-network-mmc';
+            this.networkOperatorMMC.placeholder = 'eg: 20814';
+            this.networkOperatorMMC.pattern = '[0-9]{5,6}';
+            this.networkOperatorMMC.title = 'Operator MCC/MNC';
```

Comment:
We must use the i18n keys to enable translation everywere

Reply:
I had a similar thought, but in fact, this code isn't new: it's moved from `src/plugins/Network.js`.  Maybe using i18n for it can be done in a separate PR?